treesit-auto.el

```diff
@@ -408,13 +408,43 @@ how to modify the behavior of this function."
   ;; For this mode to keep a cached copy is dangerous as it will be a global
   ;; replacement and ignores all changes while this mode is active, so
   ;; don't think it is a valid option.
-  (setq-local major-mode-remap-alist (treesit-auto--build-major-mode-remap-alist)))
+  (when (treesit-auto--enabled-on-major-mode-p global-treesit-auto-modes mode)
+    (setq-local major-mode-remap-alist (treesit-auto--build-major-mode-remap-alist))))
 
 (defun treesit-auto--on ()
   "Turn `treesit-auto-mode' on."
   (setq-local treesit-language-source-alist
               (treesit-auto--build-treesit-source-alist))
   (treesit-auto--maybe-install-grammar))
 
+(defun treesit-auto--enabled-on-major-mode-p (predicate mode)
```

Comment:
We'll need a docstring here for the new function to comply with MELPA linting.  How does this sound?

```suggestion
  "Check whether MODE should be enabled for treesit-auto.
If PREDICATE is `not', then exclude the mode from treesit-auto lists."
```